clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsMembershipManager.java

```diff
@@ -480,6 +480,7 @@ public void transitionToFatal() {
         if (previousState == MemberState.UNSUBSCRIBED) {
```

Comment:
Could you please consider modifying the unit test to keep the production code consistent?